src/data/mv_polynomial.lean

```diff
@@ -1292,4 +1323,129 @@ end
 
 end equiv
 
+/-! lemmas about `pderivative v`, the partial derivative with respect to `v : σ`-/
+section pderivative
+
+variables {R : Type} [comm_ring R]
+variable {S : Type}
+
+/-- pderivative v p is the partial derivative of p with respect to v -/
+def pderivative (v : S) (p : mv_polynomial S R) : mv_polynomial S R :=
+p.sum (λ A B, monomial (erase v A + single v (A v - 1)) (B * (A v)))
+
+lemma pderivative_monomial {a : R} {v : S} {n : ℕ} :
+  pderivative v (C a * (X v)^n) = C (a * n) * (X v)^(n - 1) :=
+begin
+  rw [←single_eq_C_mul_X, ←single_eq_C_mul_X, pderivative, monomial, sum_single_index],
+  { rw [erase_single, zero_add, single_eq_same]},
+  rw [erase_single, zero_mul, zero_add],
+  exact monomial_zero,
+end
+
+@[simp]
+lemma pderivative_C {a : R} {v : S} : pderivative v (C a) = 0 :=
+begin
+  rw [(show C a = C a * (X v)^0, by rw [pow_zero, mul_one]),
+  pderivative_monomial], simp only [mv_polynomial.C_0, nat.zero_sub, nat.cast_zero, zero_mul,
+    mul_zero, pow_zero],
+end
+
+lemma pderivative_X {v : S} : pderivative v (X v : mv_polynomial S R) = 1 :=
+begin
+  rw [(show (X v : mv_polynomial S R) = C 1 * (X v)^1,
+    by simp only [one_mul, pow_one, mv_polynomial.C_1]),
+  pderivative_monomial],
+  simp only [mul_one, nat.sub_self, nat.cast_one, mv_polynomial.C_1, pow_zero],
+end
+
+lemma pderivative_zero {v : S} : pderivative v (0: mv_polynomial S R) = 0 :=
+  finsupp.sum_zero_index
+
+lemma pderivative_monomial' {a : R} {v1 v2 : S} {n : ℕ} {hv : v1 ≠ v2} :
+  pderivative v1 (C a * (X v2)^n) = 0 :=
+begin
+  rw [←single_eq_C_mul_X, pderivative, monomial, sum_single_index],
+  {rw single_eq_of_ne, {rw [nat.cast_zero, monomial], simpa only [single_zero, mul_zero]},
+    symmetry, assumption },
```

Comment:
```suggestion
  { rw single_eq_of_ne,
    { rw [nat.cast_zero, monomial],
      simpa only [single_zero, mul_zero] },
    symmetry, assumption },
```

Reply:
I think the `symmetry, assumption` can also be given explicitly. My guess is that it is using the `h : v1 ≠ v2` assumption, right?

Reply:
Yes it is. So replace it with `exact h.symm`?